dwave/plugins/torch/boltzmann_machine.py

```diff
@@ -292,16 +299,21 @@ def sample(
                 If ``None`` and data is not a tensor then the result tensor is
                 constructed on the current device.
             sample_params (dict, optional): Parameters of the `sampler.sample` method.
+            as_tensor (bool): Whether to return the sampleset as a tensor.
+                Defaults to ``True``. If ``False`` returns a ``dimod.SampleSet``.
 
         Returns:
-            torch.Tensor: Spins sampled from the model
+            torch.Tensor | SampleSet: Spins sampled from the model
             (shape prescribed by ``sampler`` and ``sample_params``).
         """
         if sample_params is None:
             sample_params = dict()
         h, J = self.to_ising(prefactor, linear_range, quadratic_range)
-        ss = spread(sampler.sample_ising(h, J, **sample_params))
-        spins = self.sampleset_to_tensor(ss, device=device)
+        spins = spread(sampler.sample_ising(h, J, **sample_params))
+
+        if as_tensor:
+            return self.sampleset_to_tensor(spins, device=device)
+
         return spins
```

Comment:
We should be careful here with the variable names of the returned sample set. See, for example, [this snippet in sampleset_to_tensor](https://github.com/dwavesystems/dwave-pytorch-plugin/blob/main/dwave/plugins/torch/utils.py#L39-L41) and how it relabels or permutes variables to respect that of the GRBM's definition.

---

Add test to address this^